roles/qe-ssl-cert/files/ssl_cert.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+#
+# ssl_cert.sh DEST_DIR CERT_NAME CERN_CN
+#   DEST_DIR   - destination directory
+#   CERT_NAME  - base of filename of the key and csr file (eg. server)
+#   CERT_CN    - canonical name of the certificate for this key (default hostname)
+
+# ###########################################################################
+# mrg_ssl_create_key(CERT_NAME, CERN_CN)
+#   CERT_NAME  - base of filename of the key and csr file (eg. server)
+#   CERT_CN    - canonical name of the certificate for this key
+#
+#  Creates OpenSSL rsa keypair along wiht cert. signing request file (csr).
+# ###########################################################################
+function mrg_ssl_create_key() {
+  local CERT_NAME=$1
+  local CERT_CN=$2
+  test -f /etc/cron.d/redhat-ddns && \
+    local DDNS_NAME=$(head -1 /etc/cron.d/redhat-ddns | sed 's|.*name=\([-a-zA-Z0-9]*\)&.*|\1|').usersys.redhat.com
+  host ${CERT_CN} && local IP=$(host ${CERT_CN} | sed 's|[-0-9a-zA-Z.]* has address \([0-9.]*\)|\1|')
+
+
+  # Prepare openssl.cnf
+  :> /tmp/openssl.cnf
+  echo "[req]" >> /tmp/openssl.cnf
+  echo "default_bits = 2048" >> /tmp/openssl.cnf
+  echo "distinguished_name = req_distinguished_name" >> /tmp/openssl.cnf
+  echo "req_extensions = v3_ca" >> /tmp/openssl.cnf
+  echo "dirstring_type = nobmp" >> /tmp/openssl.cnf
+  echo "" >> /tmp/openssl.cnf
+  echo "[req_distinguished_name]" >> /tmp/openssl.cnf
+  echo "emailAddress = root@localhost" >> /tmp/openssl.cnf
+  echo "emailAddress_max = 40" >> /tmp/openssl.cnf
+  echo "" >> /tmp/openssl.cnf
+  echo "[v3_ca]" >> /tmp/openssl.cnf
+  echo "basicConstraints = CA:FALSE" >> /tmp/openssl.cnf
+  echo "keyUsage = nonRepudiation, digitalSignature, keyEncipherment" >> /tmp/openssl.cnf
+  echo "subjectAltName = @alt_names" >> /tmp/openssl.cnf
+  echo "" >> /tmp/openssl.cnf
+  echo "[alt_names]" >> /tmp/openssl.cnf
+  echo "DNS.1 = ${CERT_CN}" >> /tmp/openssl.cnf
+  echo "DNS.2 = $(hostname -f)" >> /tmp/openssl.cnf
```

Comment:
Could we check if dns.1 == dns.2 and add the 2nd dns name only if needed?

Reply:
@mbukatov good point! Addressed by a46639b.